apps/ui/src/composables/useActions.ts

```diff
@@ -74,7 +74,7 @@ export function useActions() {
   }
 
   async function wrapPromise(networkId: NetworkID, promise: Promise<any>) {
-    const network = getReadWriteNetwork(networkId);
+    const network = getNetwork(networkId);
```

Comment:
I'd say we should check if `receipt` has `transaction_hash` or `hash` and if doesn't just return without adding it to pending transactions.